tests/devapps/MauiApps/MauiAppB2C/Features/LogOn/B2CAuthenticationService.cs

```diff
@@ -0,0 +1,175 @@
+﻿using Microsoft.Identity.Client;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+// using Xamarin.Forms;
+
+namespace UserDetailsClient.Core.Features.LogOn
+{
+    /// <summary>
+    ///  For simplicity, we'll have this as a singleton. 
+    /// </summary>
+    public class B2CAuthenticationService
+    {
+        private readonly IPublicClientApplication _pca;
+
+
+        private static readonly Lazy<B2CAuthenticationService> lazy = new Lazy<B2CAuthenticationService>
+           (() => new B2CAuthenticationService());
+
+        public static B2CAuthenticationService Instance { get { return lazy.Value; } }
+
+
+        private B2CAuthenticationService()
+        {
+
+            // default redirectURI; each platform specific project will have to override it with its own
+            var builder = PublicClientApplicationBuilder.Create(B2CConstants.ClientID)
+                .WithB2CAuthority(B2CConstants.AuthoritySignInSignUp)
+                .WithIosKeychainSecurityGroup(B2CConstants.IOSKeyChainGroup)
+                .WithRedirectUri($"msal{B2CConstants.ClientID}://auth");
+
+            // Android implementation is based on https://github.com/jamesmontemagno/CurrentActivityPlugin
```

Comment:
You can probably create a GetParentActivityOrWindow method and use it from lines 84-91 as well